src/cascadia/QueryExtension/AzureLLMProvider.cpp

```diff
@@ -0,0 +1,209 @@
+        // there is only one case downstream from here that sets this flag to false, so start with it being true
+        bool isError{ true };
+        hstring message{};
+
+        // If the AI endpoint is not an azure open AI endpoint, return an error message
+        if (!std::regex_search(_AIEndpoint.c_str(), azureOpenAIEndpointRegex))
+        {
+            message = RS_(L"InvalidEndpointMessage");
+        }
+
+        // If we don't have a message string, that means the endpoint exists and matches the regex
+        // that we allow - now we can actually make the http request
+        if (message.empty())
+        {
+            // Make a copy of the prompt because we are switching threads
+            const auto promptCopy{ userPrompt };
+
+            // Make sure we are on the background thread for the http request
+            co_await winrt::resume_background();
+
+            WWH::HttpRequestMessage request{ WWH::HttpMethod::Post(), Uri{ _AIEndpoint } };
+            request.Headers().Accept().TryParseAdd(L"application/json");
+
+            WDJ::JsonObject jsonContent;
+            WDJ::JsonObject messageObject;
+
+            // _ActiveCommandline should be set already, we request for it the moment we become visible
+            winrt::hstring engineeredPrompt{ promptCopy };
+            if (_context && !_context.ActiveCommandline().empty())
+            {
+                engineeredPrompt = promptCopy + L". The shell I am running is " + _context.ActiveCommandline();
+            }
+            messageObject.Insert(L"role", WDJ::JsonValue::CreateStringValue(L"user"));
+            messageObject.Insert(L"content", WDJ::JsonValue::CreateStringValue(engineeredPrompt));
+            _jsonMessages.Append(messageObject);
+            jsonContent.SetNamedValue(L"messages", _jsonMessages);
+            jsonContent.SetNamedValue(L"max_tokens", WDJ::JsonValue::CreateNumberValue(800));
+            jsonContent.SetNamedValue(L"temperature", WDJ::JsonValue::CreateNumberValue(0.7));
+            jsonContent.SetNamedValue(L"frequency_penalty", WDJ::JsonValue::CreateNumberValue(0));
+            jsonContent.SetNamedValue(L"presence_penalty", WDJ::JsonValue::CreateNumberValue(0));
+            jsonContent.SetNamedValue(L"top_p", WDJ::JsonValue::CreateNumberValue(0.95));
+            jsonContent.SetNamedValue(L"stop", WDJ::JsonValue::CreateStringValue(L"None"));
+            const auto stringContent = jsonContent.ToString();
+            WWH::HttpStringContent requestContent{
+                stringContent,
+                WSS::UnicodeEncoding::Utf8,
+                L"application/json"
+            };
+
+            request.Content(requestContent);
+
+            // Send the request
+            try
+            {
+                const auto response = _httpClient.SendRequestAsync(request).get();
```

Comment:
FWIW all these `get()`s could use `co_await` in the future instead. That avoids the `resume_background` hassle.